Use WKBundlePagePostMessageWithAsyncReply for more WebKitTestRunner messages

achristensen07 opened the pull request:
#### 37e1a47538f30abbd774aad206e546afd00baa71
<pre>
Use WKBundlePagePostMessageWithAsyncReply for more WebKitTestRunner messages
<a href="https://bugs.webkit.org/show_bug.cgi?id=273211">https://bugs.webkit.org/show_bug.cgi?id=273211</a>
<a href="https://rdar.apple.com/127008368">rdar://127008368</a>

Reviewed by Sihui Liu.

This makes tests work better with site isolation because the reply is sent to the sending process
which might not be the same as the main frame process.

* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
(WTR::postMessageWithAsyncReply):
(WTR::postMessageWithAsyncReplyImpl): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setStatisticsDebugMode):
(WTR::TestRunner::setStatisticsPrevalentResourceForDebugMode):
(WTR::TestRunner::setStatisticsLastSeen):
(WTR::TestRunner::setStatisticsMergeStatistic):
(WTR::TestRunner::setStatisticsExpiredStatistic):
(WTR::TestRunner::setStatisticsPrevalentResource):
(WTR::TestRunner::setStatisticsVeryPrevalentResource):
(WTR::TestRunner::setStatisticsHasHadUserInteraction):
(WTR::TestRunner::statisticsUpdateCookieBlocking):
(WTR::TestRunner::statisticsClearInMemoryAndPersistentStoreModifiedSinceHours):
(WTR::TestRunner::statisticsClearInMemoryAndPersistentStore):
(WTR::TestRunner::statisticsClearThroughWebsiteDataRemoval):
(WTR::TestRunner::setStatisticsShouldDowngradeReferrer):
(WTR::TestRunner::setStatisticsShouldBlockThirdPartyCookies):
(WTR::TestRunner::setStatisticsFirstPartyWebsiteDataRemovalMode):
(WTR::TestRunner::statisticsSetToSameSiteStrictCookies):
(WTR::TestRunner::statisticsSetFirstPartyHostCNAMEDomain):
(WTR::TestRunner::statisticsSetThirdPartyCNAMEDomain):
(WTR::TestRunner::statisticsResetToConsistentState):
(WTR::TestRunner::statisticsCallDidSetPrevalentResourceForDebugModeCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetLastSeenCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetMergeStatisticCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetExpiredStatisticCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetPrevalentResourceCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetVeryPrevalentResourceCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetHasHadUserInteractionCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetBlockCookiesForHostCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetShouldDowngradeReferrerCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetShouldBlockThirdPartyCookiesCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetFirstPartyWebsiteDataRemovalModeCallback): Deleted.
(WTR::TestRunner::statisticsCallClearInMemoryAndPersistentStoreCallback): Deleted.
(WTR::TestRunner::statisticsCallClearThroughWebsiteDataRemovalCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetToSameSiteStrictCookiesCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetFirstPartyHostCNAMEDomainCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetThirdPartyCNAMEDomainCallback): Deleted.
(WTR::TestRunner::statisticsCallDidResetToConsistentStateCallback): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::adoptAndCallCompletionHandler):
(WTR::TestController::setStatisticsDebugMode):
(WTR::TestController::setStatisticsPrevalentResourceForDebugMode):
(WTR::TestController::setStatisticsLastSeen):
(WTR::TestController::setStatisticsMergeStatistic):
(WTR::TestController::setStatisticsExpiredStatistic):
(WTR::TestController::setStatisticsPrevalentResource):
(WTR::TestController::setStatisticsVeryPrevalentResource):
(WTR::TestController::setStatisticsHasHadUserInteraction):
(WTR::TestController::statisticsUpdateCookieBlocking):
(WTR::TestController::statisticsClearInMemoryAndPersistentStore):
(WTR::TestController::statisticsClearInMemoryAndPersistentStoreModifiedSinceHours):
(WTR::TestController::statisticsClearThroughWebsiteDataRemoval):
(WTR::TestController::setStatisticsShouldDowngradeReferrer):
(WTR::TestController::setStatisticsShouldBlockThirdPartyCookies):
(WTR::TestController::setStatisticsFirstPartyWebsiteDataRemovalMode):
(WTR::TestController::setStatisticsToSameSiteStrictCookies):
(WTR::TestController::setStatisticsFirstPartyHostCNAMEDomain):
(WTR::TestController::setStatisticsThirdPartyCNAMEDomain):
(WTR::TestController::statisticsResetToConsistentState):
(WTR::TestController::removeAllCookies):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didClearStatisticsInMemoryAndPersistentStore): Deleted.
(WTR::TestInvocation::didClearStatisticsThroughWebsiteDataRemoval): Deleted.
(WTR::TestInvocation::didSetShouldDowngradeReferrer): Deleted.
(WTR::TestInvocation::didSetShouldBlockThirdPartyCookies): Deleted.
(WTR::TestInvocation::didSetFirstPartyWebsiteDataRemovalMode): Deleted.
(WTR::TestInvocation::didSetToSameSiteStrictCookies): Deleted.
(WTR::TestInvocation::didSetFirstPartyHostCNAMEDomain): Deleted.
(WTR::TestInvocation::didSetThirdPartyCNAMEDomain): Deleted.
(WTR::TestInvocation::didResetStatisticsToConsistentState): Deleted.
(WTR::TestInvocation::didSetBlockCookiesForHost): Deleted.
(WTR::TestInvocation::didSetPrevalentResourceForDebugMode): Deleted.
(WTR::TestInvocation::didSetLastSeen): Deleted.
(WTR::TestInvocation::didMergeStatistic): Deleted.
(WTR::TestInvocation::didSetExpiredStatistic): Deleted.
(WTR::TestInvocation::didSetPrevalentResource): Deleted.
(WTR::TestInvocation::didSetVeryPrevalentResource): Deleted.
(WTR::TestInvocation::didSetHasHadUserInteraction): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:

Canonical link: <a href="https://commits.webkit.org/277962@main">https://commits.webkit.org/277962@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f15ee0ea7f966b4690c7c6008c86f1b72e892a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49082 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28321 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52062 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51825 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45158 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51386 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34291 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25853 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40106 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50535 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25925 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42305 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21219 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23380 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43478 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7276 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/45308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43983 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53739 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/24158 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20366 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47425 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/25437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42511 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46396 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/26223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7029 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/25157 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
```diff
@@ -1087,22 +1002,22 @@ static void postMessageWithAsyncReplyImpl(const char* name, WKRetainPtr<WKTypeRe
 }
 
-void postMessageWithAsyncReply(const char* name, bool value, JSValueRef callback)
+void postMessageWithAsyncReply(const char* messageName, JSStringRef value, JSValueRef callback)
```
Seems unnecessary to add a specific function for `JSStringRef` if client needs to call `toWK` for other types?

szewai reviewed (approved):
r=me if bots are happy

achristensen07 committed:
Use WKBundlePagePostMessageWithAsyncReply for more WebKitTestRunner messages
https://bugs.webkit.org/show_bug.cgi?id=273211
rdar://127008368

Reviewed by Sihui Liu.

This makes tests work better with site isolation because the reply is sent to the sending process
which might not be the same as the main frame process.

* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
(WTR::postMessageWithAsyncReply):
(WTR::postMessageWithAsyncReplyImpl): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setStatisticsDebugMode):
(WTR::TestRunner::setStatisticsPrevalentResourceForDebugMode):
(WTR::TestRunner::setStatisticsLastSeen):
(WTR::TestRunner::setStatisticsMergeStatistic):
(WTR::TestRunner::setStatisticsExpiredStatistic):
(WTR::TestRunner::setStatisticsPrevalentResource):
(WTR::TestRunner::setStatisticsVeryPrevalentResource):
(WTR::TestRunner::setStatisticsHasHadUserInteraction):
(WTR::TestRunner::statisticsUpdateCookieBlocking):
(WTR::TestRunner::statisticsClearInMemoryAndPersistentStoreModifiedSinceHours):
(WTR::TestRunner::statisticsClearInMemoryAndPersistentStore):
(WTR::TestRunner::statisticsClearThroughWebsiteDataRemoval):
(WTR::TestRunner::setStatisticsShouldDowngradeReferrer):
(WTR::TestRunner::setStatisticsShouldBlockThirdPartyCookies):
(WTR::TestRunner::setStatisticsFirstPartyWebsiteDataRemovalMode):
(WTR::TestRunner::statisticsSetToSameSiteStrictCookies):
(WTR::TestRunner::statisticsSetFirstPartyHostCNAMEDomain):
(WTR::TestRunner::statisticsSetThirdPartyCNAMEDomain):
(WTR::TestRunner::statisticsResetToConsistentState):
(WTR::TestRunner::statisticsCallDidSetPrevalentResourceForDebugModeCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetLastSeenCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetMergeStatisticCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetExpiredStatisticCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetPrevalentResourceCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetVeryPrevalentResourceCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetHasHadUserInteractionCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetBlockCookiesForHostCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetShouldDowngradeReferrerCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetShouldBlockThirdPartyCookiesCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetFirstPartyWebsiteDataRemovalModeCallback): Deleted.
(WTR::TestRunner::statisticsCallClearInMemoryAndPersistentStoreCallback): Deleted.
(WTR::TestRunner::statisticsCallClearThroughWebsiteDataRemovalCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetToSameSiteStrictCookiesCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetFirstPartyHostCNAMEDomainCallback): Deleted.
(WTR::TestRunner::statisticsCallDidSetThirdPartyCNAMEDomainCallback): Deleted.
(WTR::TestRunner::statisticsCallDidResetToConsistentStateCallback): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::adoptAndCallCompletionHandler):
(WTR::TestController::setStatisticsDebugMode):
(WTR::TestController::setStatisticsPrevalentResourceForDebugMode):
(WTR::TestController::setStatisticsLastSeen):
(WTR::TestController::setStatisticsMergeStatistic):
(WTR::TestController::setStatisticsExpiredStatistic):
(WTR::TestController::setStatisticsPrevalentResource):
(WTR::TestController::setStatisticsVeryPrevalentResource):
(WTR::TestController::setStatisticsHasHadUserInteraction):
(WTR::TestController::statisticsUpdateCookieBlocking):
(WTR::TestController::statisticsClearInMemoryAndPersistentStore):
(WTR::TestController::statisticsClearInMemoryAndPersistentStoreModifiedSinceHours):
(WTR::TestController::statisticsClearThroughWebsiteDataRemoval):
(WTR::TestController::setStatisticsShouldDowngradeReferrer):
(WTR::TestController::setStatisticsShouldBlockThirdPartyCookies):
(WTR::TestController::setStatisticsFirstPartyWebsiteDataRemovalMode):
(WTR::TestController::setStatisticsToSameSiteStrictCookies):
(WTR::TestController::setStatisticsFirstPartyHostCNAMEDomain):
(WTR::TestController::setStatisticsThirdPartyCNAMEDomain):
(WTR::TestController::statisticsResetToConsistentState):
(WTR::TestController::removeAllCookies):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didClearStatisticsInMemoryAndPersistentStore): Deleted.
(WTR::TestInvocation::didClearStatisticsThroughWebsiteDataRemoval): Deleted.
(WTR::TestInvocation::didSetShouldDowngradeReferrer): Deleted.
(WTR::TestInvocation::didSetShouldBlockThirdPartyCookies): Deleted.
(WTR::TestInvocation::didSetFirstPartyWebsiteDataRemovalMode): Deleted.
(WTR::TestInvocation::didSetToSameSiteStrictCookies): Deleted.
(WTR::TestInvocation::didSetFirstPartyHostCNAMEDomain): Deleted.
(WTR::TestInvocation::didSetThirdPartyCNAMEDomain): Deleted.
(WTR::TestInvocation::didResetStatisticsToConsistentState): Deleted.
(WTR::TestInvocation::didSetBlockCookiesForHost): Deleted.
(WTR::TestInvocation::didSetPrevalentResourceForDebugMode): Deleted.
(WTR::TestInvocation::didSetLastSeen): Deleted.
(WTR::TestInvocation::didMergeStatistic): Deleted.
(WTR::TestInvocation::didSetExpiredStatistic): Deleted.
(WTR::TestInvocation::didSetPrevalentResource): Deleted.
(WTR::TestInvocation::didSetVeryPrevalentResource): Deleted.
(WTR::TestInvocation::didSetHasHadUserInteraction): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:

Canonical link: https://commits.webkit.org/277962@main
